### PR TITLE
Revert and re-adapt test/e2e/common/utils.py

### PR DESCRIPTION
**Pull request by israel-hdez**
**What this PR does / why we need it**:

This reverts changes done in commit ecff079 to test/e2e/common/utils.py. It is possible to revert since opendatahub-io/odh-model-controller#59 is solved.

After the revert, another adaptation was needed. CI in upstream assumes that the URL in the InferenceService cannot be resolved, so it uses the IP of the istio-ingressgateway together with the HTTP Host header to workaround this limitation. However, in ODH testing environment the URL of InferenceServices can be resolved. The adaptation adds a flag to use the URL of the isvc, rather than using the IP of the istio-ingressgateway.

This adaptation can be contributed to upstream, because the community can benefit from it.

The CI setup is updated, so that the revert works correctly:
* odh-model-controller is now being installed
* TLS is removed from the istio-ingressgateway

Additionally, the following other changes were done to the CI setup:
* Turn off Mesh-related components that are not needed for CI
* Prefer usage of the ServiceMeshMember CRD to enroll namespaces to the Mesh

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes opendatahub-io/kserve#50

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/common/utils.py [hunk 1]
```diff
@@ -244,3 +238,13 @@ def predict_modelmesh(service_name, input_json, pod_name, model_name=None):
...
+
+
+def get_isvc_endpoint(isvc):
+    host = urlparse(isvc["status"]["url"]).netloc
+    path = urlparse(isvc["status"]["url"]).path
+    if os.environ.get("CI_USE_ISVC_HOST") == "1":
+        cluster_ip = host
+    else:
+        cluster_ip = get_cluster_ip()
+    return cluster_ip, host, path
```
Comment (spolti) [inline test/e2e/common/utils.py:250]: new line

File: test/scripts/openshift-ci/deploy.serverless.sh [hunk 2]
```diff
@@ -144,12 +147,9 @@ spec:
     - hosts:
         - '*'
       port:
-        name: https
-        number: 443
```
Comment (spolti) [inline test/scripts/openshift-ci/deploy.serverless.sh:148]: isn't better to keep ssl to test self signed certificates?

File: test/e2e/common/utils.py (hunk 1 above)
Comment (israel-hdez) [reply test/e2e/common/utils.py:250]: New line before `return`?

File: test/scripts/openshift-ci/deploy.serverless.sh (hunk 2 above)
Comment (israel-hdez) [reply test/scripts/openshift-ci/deploy.serverless.sh:148]: I'd like to say yes, but the upstream E2E tests don't work with TLS.

We probably should contribute a basic test suite that has ensures self-signed TLS works.

Comment (israel-hdez) [reply test/scripts/openshift-ci/deploy.serverless.sh:148]: Thinking twice, testing TLS here doesn't make too much sense. The point where makes sense in the storage initializer fetching from self signed storage.

File: test/e2e/common/utils.py [hunk 3]
```diff
@@ -154,8 +150,7 @@ def explain_response(service_name, input_json):
     )
     # temporary sleep until this is fixed https://github.com/kserve/kserve/issues/604
     time.sleep(10)
```
Comment (dtrifiro) [inline test/e2e/common/utils.py:152]: https://github.com/kserve/kserve/issues/604 has been merged, can we remove this or are we waiting on something else?

File: test/e2e/common/utils.py (hunk 1 above)
Comment (spolti) [reply test/e2e/common/utils.py:250]: at the end of the file :)

File: test/e2e/common/utils.py (hunk 3 above)
Comment (israel-hdez) [reply test/e2e/common/utils.py:152]: IDK. Probably, there is no need to way.
We should contribute such change directly to upstream.

File: test/scripts/openshift-ci/run-e2e-tests.sh [hunk 4]
```diff
@@ -77,6 +77,9 @@ kustomize build $PROJECT_ROOT/config/overlays/test | \
   oc apply -f -
 oc wait --for=condition=ready pod -l control-plane=kserve-controller-manager -n kserve --timeout=300s
 
+echo "Installing odh-model-controller"
+kustomize build $PROJECT_ROOT/test/scripts/openshift-ci | oc apply -f -
```
Comment (terrytangyuan) [inline test/scripts/openshift-ci/run-e2e-tests.sh:81]: This can be simplified `oc apply -k $PROJECT_ROOT/test/scripts/openshift-ci`

Comment (israel-hdez) [reply test/scripts/openshift-ci/run-e2e-tests.sh]: Fixed.

File: test/e2e/common/utils.py (hunk 1 above)
Comment (israel-hdez) [reply test/e2e/common/utils.py:250]: Fixed.

Comment (israel-hdez) [reply test/e2e/common/utils.py:250]: Sorry, I reverted.
The python linter didn't like it. See: https://github.com/opendatahub-io/kserve/actions/runs/7269665250/job/19807632226#step:4:38

Comment (spolti) [reply test/e2e/common/utils.py:250]: hmm, not sure why.
PEP8 suggests a line terminator while this custom linter asks to remove.

**Review by spolti — APPROVED** [review]

**Pull request merged**